### PR TITLE
Build AMGX 2.3 with CUDA platform

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: A/AMGX/build_tarballs.jl [hunk 1]
```diff
@@ -55,9 +62,21 @@ products = [
...
+
+for cuda_version in [v"11.0"], platform in platforms
+    augmented_platform = Platform(arch(platform), os(platform); cuda=CUDA.platform(cuda_version))
+    should_build_platform(triplet(augmented_platform)) || continue
+
+    cuda_deps = [
+        BuildDependency(PackageSpec(name="CUDA_full_jll",
+                                    version=cuda_full_versions[cuda_version])),
+        RuntimeDependency(PackageSpec(name="CUDA_Runtime_jll",
+                                      version=v"0.2"), compat="0.2"),  # avoid pulling in CUDA 12 for now.
```
Comment (maleadt) [inline A/AMGX/build_tarballs.jl:75]: No need for a version bound. AMGX.jl should check if AMGX_jll.is_available(), and if not, notify that e.g. this platform or CUDA version is unsupported (which can be remedied by setting a preference, by calling CUDA.set_runtime_version!)

Comment (vchuravy) [reply A/AMGX/build_tarballs.jl:75]: I took this from Libxc btw

File: A/AMGX/build_tarballs.jl [hunk 2]
```diff
@@ -55,9 +62,20 @@ products = [
     LibraryProduct("libamgxsh", :libamgxsh),
 ]
 
-dependencies = [
-    BuildDependency(PackageSpec(name = "CUDA_full_jll", version = "10.0")),
-    Dependency(PackageSpec(name = "CUDA_jll", version = "10.0")),
-]
+dependencies = [RuntimeDependency(PackageSpec(name="CUDA_Runtime_jll"))]
```
Comment (maleadt) [inline A/AMGX/build_tarballs.jl:65]: `CUDA_Runtime_jll` listed twice?

**Comment by vchuravy** [comment]
When I turn the audit on locally I get:

```
ERROR: could not load library "/home/vchuravy/builds/Yggdrasil/A/AMGX/build/x86_64-linux-gnu-cuda+10.2/QZzyoaND/x86_64-linux-gnu-libgfortran3-cxx03-cuda+10.2/destdir/lib/libamgxsh.so"
libcublasLt.so.10: cannot open shared object file: No such file or directory
```

**Pull request merged**